oracle/datadog_checks/oracle/oracle.py

```diff
@@ -47,43 +50,75 @@ class Oracle(AgentCheck):
-            are reachable.
-            """
-            self.log.error(msg)
-            return
-
-        self.log.debug('Running cx_Oracle version {0}'.format(cx_Oracle.version))
-        server, user, password, service, tags = self._get_config(instance)
+        self.use_oracle_client = True
+        try:
+            # Check if the instantclient is available
+            cx_Oracle.clientversion()
+            self.log.debug('Running cx_Oracle version {0}'.format(cx_Oracle.version))
+        except cx_Oracle.DatabaseError, e:
+            # Fallback to JDBC
+            self.use_oracle_client = False
+            self.log.info('Oracle instant client unavailable, falling back to JDBC: {}'.format(e))
+
+        server, user, password, service, jdbc_driver, tags = self._get_config(instance)
 
         if not server or not user:
             raise Exception("Oracle host and user are needed")
 
-        con = self._get_connection(server, user, password, service, tags)
+        con = self._get_connection(server, user, password, service, jdbc_driver, tags)
 
         self._get_sys_metrics(con, tags)
         self._get_tablespace_metrics(con, tags)
 
+        con.close()
+
     def _get_config(self, instance):
         self.server = instance.get('server', None)
         user = instance.get('user', None)
         password = instance.get('password', None)
         service = instance.get('service_name', None)
+        jdbc_driver = instance.get('jdbc_driver_path', None)
         tags = instance.get('tags', [])
-        return (self.server, user, password, service, tags)
+        return (self.server, user, password, service, jdbc_driver, tags)
 
-    def _get_connection(self, server, user, password, service, tags):
+    def _get_connection(self, server, user, password, service, jdbc_driver, tags):
         if tags is None:
             tags = []
         self.service_check_tags = [
             'server:%s' % server
         ] + tags
-        connect_string = '{0}/{1}@//{2}/{3}'.format(user, password, server, service)
+
+        if self.use_oracle_client:
+            connect_string = CX_CONNECT_STRING.format(user, password, server, service)
+        else:
+            connect_string = JDBC_CONNECT_STRING.format(server, service)
+
         try:
-            con = cx_Oracle.connect(connect_string)
+            if self.use_oracle_client:
+                con = cx_Oracle.connect(connect_string)
+            else:
+                try:
+                    if jpype.isJVMStarted() and not jpype.isThreadAttachedToJVM():
+                        jpype.attachThreadToJVM()
+                        jpype.java.lang.Thread.currentThread().setContextClassLoader(
+                            jpype.java.lang.ClassLoader.getSystemClassLoader())
+                    con = jdb.connect(ORACLE_DRIVER_CLASS, connect_string, [user, password], jdbc_driver)
+                except jpype.JException(jpype.java.lang.RuntimeException), e:
```

Comment:
Since this syntax is deprecated, let's do `except jpype.JException(jpype.java.lang.RuntimeException) as e:` instead.